### ISSUE TITLE
detect mingw in source file

### ISSUE THREAD

**Issue by starwing** [lunarmodules/luasocket#67]
Hi,

if you compile luasocket without makefile, you must define LUASOCKET_INET_PTON, so why not we make things easier? this is my patch:

```
diff --git a/src/inet.h b/src/inet.h
index 1f1a96a..55cf61c 100644
--- a/src/inet.h
+++ b/src/inet.h
@@ -22,6 +22,10 @@
 #define LUASOCKET_INET_ATON
 #endif

+#ifdef __MINGW32__
+#define LUASOCKET_INET_PTON
+#endif
+
 int inet_open(lua_State *L);

 const char *inet_trycreate(p_socket ps, int family, int type);
```

this makes compile easier.

btw, it's better to add a .def file to export all symbols in windows, thus we needn't define __declspec(export) anymore.

**Comment by ewestbrook** [comment]
In general, we rely on the presence of `_WIN32` to detect Windows builds (as is commonly done).

In Windows builds, LuaSocket uses the `__declspec(dllexport)` directive for the few `LUASOCKET_API` functions that need to be exported by LuaSocket.

In this particular case, `inet.h` uses the `_WIN32` condition to define `LUASOCKET_INET_PTON` for all Windows builds -- including MinGW.

Does this satisfy your case?  Please add a response if further action is needed, otherwise it will be assumed that all is well.

**Comment by alerque** [comment]
The rockspec's use of `LUASOCKET_INET_PTON` was changed in #383 and released in v3.1.0. If some Windows folks from here can look into this and identify if changes need to be made to the makefile too I would be willing to facilitate a PR to get this issue closed.

**Comment by alerque** [comment]
It seems like this issue might be resolved in recent releases. If anybody has evidence to the contrary feel free to pipe up with a comment and I'll re-evaluate.
